lib/gollum/app.rb

```diff
@@ -355,6 +355,9 @@ class App < Sinatra::Base
         if settings.wiki_options[:template_page] then
           temppage = wiki_page('/_Template')
           @template_page = (temppage.page != nil) ? temppage.page.raw_data : 'Template page option is set, but no /_Template page is present or committed.'
+          if defined?(Gollum::TemplateFilter)
+            @template_page = Gollum::TemplateFilter.filter(@template_page)
```

Comment:
Quite a lot happening here in the middle of a Sinatra route. Would it make sense to:

1. Change `TemplateFilter` to (e.g.) `TemplatePage`
2. Add the logic for checking for/loading the template data to the `TemplatePage` class (for instance, to the `.load` method)
3. Simply call `@template_page = TemplatePage.load` here

Reply:
Also, another approach to the API could be the following. Perhaps it would be more user friendly? Not sure.

```ruby
TemplatePage.add_filter('{{current_date}}') do
  Time.now.to_s
end
```

The add filter method would do something like:

For example:
```ruby
def self.add_filter(pattern, &replacement)
  @@filters[pattern] = replacement
end
```
And then in the `filter` method:

```ruby
@@template_filter.each |pattern, replacement|
data.gsub!(pattern, replacement.call)
end
```